### PR TITLE
This adds tests for models, views and forms in the project

### PR DESCRIPTION
**Pull request by hamadhassan3**
# Overview of the work
This project is part of the 3rd week in python training program. It implements a website based on the Django web framework.

# Major changes

1. Unit tests for some of the models, views and forms are added now.
2. A minor fix in label for date of death in author model.

# Output
tests

# Feedback Required
I need feedback on the quality and style of the code. Please also mention if any part of the code can be improved. Most of the test cases were already provided, so the major addition is the CreateAuthorViewTest class in test_views.py in tests module. Please also comment on the quality of these test cases.

### CODE REVIEW

File: catalog/tests/test_forms.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import datetime
+
+from django.test import TestCase
+from django.utils import timezone
+
+from catalog.forms import RenewBookForm
+
+class RenewBookFormTest(TestCase):
+
+
```
Comment (shahabejaz) [inline catalog/tests/test_forms.py]: Why adding this gap? Is this something in tutorial?

File: catalog/tests/test_views.py [hunk 2]
```diff
@@ -0,0 +1,351 @@
+import datetime
+import uuid
+
```
Comment (shahabejaz) [inline catalog/tests/test_views.py]: I think this space in between is unnecessary. Same for other places where there is space in between imports.

File: catalog/tests/test_views.py [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+        response = self.client.get(reverse('authors')+'?page=2')
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue('is_paginated' in response.context)
+        self.assertTrue(response.context['is_paginated'] == True)
+        self.assertEqual(len(response.context['author_list']), 3)
+
+
+
+
+
```
Comment (shahabejaz) [inline catalog/tests/test_views.py]: This space should be consistent. We usually leave two line space between classes.

File: catalog/tests/test_views.py [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+        self.assertEqual(str(response.context['user']), 'testuser1')
+        # Check that we got a response "success"
+        self.assertEqual(response.status_code, 200)
+
+        # Check we used correct template
+        self.assertTemplateUsed(response, 'catalog/bookinstance_list_borrowed_user.html')
+
+
+
+
```
Comment (shahabejaz) [inline catalog/tests/test_views.py]: You can reduce it to two line.

**Review by shahabejaz — COMMENTED** [review]
There are some issues related to formatting but we usually resolve those by using linters.

File: catalog/tests/test_forms.py (hunk 1 above)
Comment (hamadhassan3) [reply catalog/tests/test_forms.py]: I was about to add a comment here but forgot about it. I have added comment here now.

File: catalog/tests/test_views.py (hunk 2 above)
Comment (hamadhassan3) [reply catalog/tests/test_views.py]: Some code was copied from the tutorial here and I forgot to ensure that it used proper styling strategies. It is fixed now. I also googled a bit on linters.

File: catalog/tests/test_views.py (hunk 3 above)
Comment (hamadhassan3) [reply catalog/tests/test_views.py]: It is fixed now

File: catalog/tests/test_views.py (hunk 4 above)
Comment (hamadhassan3) [reply catalog/tests/test_views.py]: It is fixed now

File: .gitignore [hunk 5]
```diff
@@ -1,3 +1,8 @@
...
-catalog/migrations/__pycache__/
-locallibrary/__pycache__/
+# pyc files
+*.pyc
+
+# Text backup files
+*.bak
+
+# Database
+*.sqlite3
```
Comment (shahabejaz) [inline .gitignore:8]: 👍

**Review by shahabejaz — APPROVED** [review]

**Pull request merged**